omegaconf/base.py

```diff
@@ -377,28 +400,63 @@ def _resolve_interpolation(
         elif value_kind == ValueKind.STR_INTERPOLATION:
             value = _get_value(value)
             assert isinstance(value, str)
-            orig = value
-            new = ""
-            last_index = 0
-            for match in match_list:
-                new_val = self._resolve_simple_interpolation(
+            try:
+                # NB: `match_list` is ignored here as complex interpolations may be
+                # nested, the string will be re-parsed within `_evaluate_complex()`.
+                new = self._evaluate_complex(
+                    value,
                     key=key,
-                    inter_type=match.group(1),
-                    inter_key=match.group(2),
                     throw_on_missing=throw_on_missing,
                     throw_on_resolution_failure=throw_on_resolution_failure,
                 )
-                # if failed to resolve, return None for the whole thing.
-                if new_val is None:
-                    return None
-                new += orig[last_index : match.start(0)] + str(new_val)
-                last_index = match.end(0)
-
-            new += orig[last_index:]
+            except ParseError:
+                return None
```

Comment:
It it fly, if there is a need to suppress it in some scenarios we can add another flag.

Reply:
Changed in fb08218

I still hard to guard the exception within an `if throw_on_resolution_failure:` block as otherwise there would be an infinite recursion on error (because the code preparing the error message also tries to parse the same expression)